### PR TITLE
JCRVLT-332 - adding support to JcrPackageRegistry to resolve dependen…

### PR DESCRIPTION
**Pull request by DominikSuess**
…cies from FsPackageRegistry

### CODE REVIEW

File: vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/registry/impl/JcrPackageRegistry.java [hunk 1]
```diff
@@ -115,6 +115,11 @@
      * the package root prefix of the primary root path.
      */
     private final String primaryPackRootPathPrefix;
+    
+    /**
+     * FSPackageRegistry can be registered if present in the system to be able to look up presatisfied dependencies
+     */
+    private FSPackageRegistry fsPackageRegistry = null;
```
Comment (tripodsan) [inline vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/registry/impl/JcrPackageRegistry.java]: does this need to be a `FSPackageRegistry` or could this just be a _base_ or _fallback_ package registry?

```suggestion
    private PackageRegistry baseRegistry = null;
```

File: vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/registry/impl/JcrPackageRegistry.java [hunk 2]
```diff
@@ -241,6 +249,11 @@ public Node getPrimaryPackageRoot(boolean autoCreate) throws RepositoryException
     public RegisteredPackage open(@Nonnull PackageId id) throws IOException {
         try {
             Node node = getPackageNode(id);
+            if (node == null && baseRegistry != null) {
+                if (baseRegistry.contains(id)) {
```
Comment (tripodsan) [inline vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/registry/impl/JcrPackageRegistry.java]: can't this be simplified to just:

```suggestion
                return baseRegistry.open(id)
```

or does it bring additional value to check `contains()` first?

**Comment by DominikSuess** [comment]
public JcrPackage open(Node node, boolean allowInvalid);
won't work as Registry doesn't return a JcrPackage but a RegisteredPackage. Contains I'll be adjusting

**Comment by DominikSuess** [comment]
Pushed requested changes but the open method of JcrPackageRegistry which explicitly opens a JcrPackage which the Registry Interface is not aware of (and in FSRegistry wouldn't exist)

**Pull request closed without merging**